File: C#/WindowsPhone/LGame-XNA-lib/Loon.Foundation/NSMutableArray.cs
using Loon.Utils.Collection;
namespace Loon.Foundation
{

    public class NSMutableArray : NSArray
    {

        public static NSMutableArray Array()
        {
            return new NSMutableArray();
        }

        public new static NSMutableArray ArrayWithObject(NSObject o)
        {
            return new NSMutableArray(o);
        }

        public new static NSMutableArray ArrayWithObjects(params NSObject[] objects)
        {
            return new NSMutableArray(objects);
        }

        public new static NSMutableArray ArrayWithArray(NSArray array)
        {
            return new NSMutableArray(array);
        }

        public static NSMutableArray ArrayWithCapacity(int capacity)
        {
            return new NSMutableArray(capacity);
        }

        public NSMutableArray(NSArray array)
        {
            this._list = array._list;
        }

        public NSMutableArray(params NSObject[] objects)
        {
            _list = new ArrayList(objects.Length);
            foreach (NSObject o in objects)
            {
                _list.Add(o);
            }
        }

        public NSMutableArray(NSObject o)
        {
            _list = new ArrayList(1);
            _list.Add(o);
        }

        public NSMutableArray(int capacity)
        {
            _list = new ArrayList(capacity);
        }

        public NSMutableArray()
        {
            _list = new ArrayList();
        }

        public void RemoveAllObjects()
        {
            _list.Clear();
        }

        public void AddObjectsFromArray(NSArray a)
        {
            _list.AddAll(a._list);
        }

        public void ReplaceObject(int index, NSObject o)
        {
            _list.Set(index, o);
        }
    }
}

File: Loon-0.5-Alpha/src/loon/foundation/NSMutableArray.java
/**
 * Copyright 2013 The Loon Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package loon.foundation;

import loon.utils.collection.ArrayList;

public class NSMutableArray extends NSArray {

	public static NSMutableArray array() {
		return new NSMutableArray();
	}

	public static NSMutableArray arrayWithObject(NSObject o) {
		return new NSMutableArray(o);
	}

	public static NSMutableArray arrayWithObjects(NSObject... objects) {
		return new NSMutableArray(objects);
	}

	public static NSMutableArray arrayWithArray(NSArray array) {
		return new NSMutableArray(array);
	}

	public static NSMutableArray arrayWithCapacity(int capacity) {
		return new NSMutableArray(capacity);
	}

	public NSMutableArray(NSArray array) {
		this._list = array._list;
	}

	public NSMutableArray(NSObject... objects) {
		_list = new ArrayList(objects.length);
		for (NSObject o : objects) {
			_list.add(o);
		}
	}

	public NSMutableArray(NSObject o) {
		_list = new ArrayList(1);
		_list.add(o);
	}

	public NSMutableArray(int capacity) {
		_list = new ArrayList(capacity);
	}

	public NSMutableArray() {
		_list = new ArrayList();
	}

	public void removeAllObjects() {
		_list.clear();
	}

	public void addObjectsFromArray(NSArray a) {
		_list.addAll(a._list);
	}

	public void replaceObject(int index, NSObject o) {
		_list.set(index, o);
	}
}
